File: point-rewards-frontend/src/lib/encryption.ts
import CryptoJS from 'crypto-js';

// 加密密钥 - 注意：这在前端是不安全的，仅用于满足你的需求
const SECRET_KEY = 'eternalmoon';

/**
 * 生成AES密钥 (与后端Python保持一致)
 */
function getKey(): CryptoJS.lib.WordArray {
  return CryptoJS.SHA256(SECRET_KEY);
}

/**
 * AES加密 (兼容Python后端的CBC模式)
 * @param text 要加密的文本
 * @returns 加密后的字符串
 */
export function encryptPassword(text: string): string {
  try {
    const key = getKey();
    const iv = CryptoJS.lib.WordArray.random(16); // 16字节IV
    
    const encrypted = CryptoJS.AES.encrypt(text, key, {
      iv: iv,
      mode: CryptoJS.mode.CBC,
      padding: CryptoJS.pad.Pkcs7
    });
    
    // 组合IV和加密数据，然后转换为base64
    const combined = iv.concat(encrypted.ciphertext);
    return CryptoJS.enc.Base64.stringify(combined);
  } catch (error) {
    console.error('加密失败:', error);
    throw new Error('密码加密失败');
  }
}

/**
 * AES解密 (兼容Python后端的CBC模式)
 * @param encryptedText 加密的文本
 * @returns 解密后的字符串
 */
export function decryptPassword(encryptedText: string): string {
  try {
    const key = getKey();
    
    // 解码base64
    const combined = CryptoJS.enc.Base64.parse(encryptedText);
    
    // 提取IV和加密数据
    const iv = CryptoJS.lib.WordArray.create(combined.words.slice(0, 4)); // 前16字节作为IV
    const encrypted = CryptoJS.lib.WordArray.create(combined.words.slice(4)); // 剩余作为加密数据
    
    const decrypted = CryptoJS.AES.decrypt(
      CryptoJS.lib.CipherParams.create({
        ciphertext: encrypted
      }),
      key,
      {
        iv: iv,
        mode: CryptoJS.mode.CBC,
        padding: CryptoJS.pad.Pkcs7
      }
    );
    
    const decryptedText = decrypted.toString(CryptoJS.enc.Utf8);
    if (!decryptedText) {
      throw new Error('解密失败');
    }
    return decryptedText;
  } catch (error) {
    console.error('解密失败:', error);
    throw new Error('密码解密失败');
  }
}

/**
 * 判断密码是否已加密
 * @param password 密码字符串
 * @returns 是否已加密
 */
export function isEncryptedPassword(password: string): boolean {
  try {
    if (password.length < 20) {
      return false;
    }
    // 尝试base64解码
    CryptoJS.enc.Base64.parse(password);
    return true;
  } catch {
    return false;
  }
}

File: point-rewards-backend/app/encryption.py
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import hashlib

# 与前端相同的密钥
SECRET_KEY = 'eternalmoon'

def get_key():
    """生成AES密钥"""
    return hashlib.sha256(SECRET_KEY.encode()).digest()

def encrypt_password(password: str) -> str:
    """
    AES加密密码
    注意：这个函数主要用于测试，生产环境应该在前端加密
    """
    try:
        key = get_key()
        cipher = AES.new(key, AES.MODE_CBC)
        
        # 填充数据
        padded_data = pad(password.encode(), AES.block_size)
        
        # 加密
        encrypted = cipher.encrypt(padded_data)
        
        # 组合IV和加密数据
        result = base64.b64encode(cipher.iv + encrypted).decode()
        return result
    except Exception as e:
        raise Exception(f"加密失败: {str(e)}")

def decrypt_password(encrypted_password: str) -> str:
    """
    AES解密密码
    """
    try:
        key = get_key()
        
        # 解码base64
        encrypted_data = base64.b64decode(encrypted_password.encode())
        
        # 提取IV和加密数据
        iv = encrypted_data[:16]
        encrypted = encrypted_data[16:]
        
        # 解密
        cipher = AES.new(key, AES.MODE_CBC, iv)
        decrypted = cipher.decrypt(encrypted)
        
        # 去除填充
        password = unpad(decrypted, AES.block_size).decode()
        return password
    except Exception as e:
        raise Exception(f"解密失败: {str(e)}")

def is_encrypted_password(password: str) -> bool:
    """
    判断密码是否已加密
    简单判断：加密后的密码通常较长且包含base64字符
    """
    try:
        if len(password) < 20:  # 加密后的密码应该比较长
            return False
        # 尝试base64解码
        base64.b64decode(password.encode())
        return True
    except:
        return False